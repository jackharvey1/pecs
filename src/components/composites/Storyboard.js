import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import StoryboardImage from './StoryboardImage';
import Adder from '../atomics/Adder';

const FlexRow = styled.div`
    position: relative;

    display: flex;
    align-items: center;
    flex-direction: row;

    overflow-x: auto;
    overflow-y: hidden;

    width: 100vw;
    height: calc(100vh - 40px);

    --spacing: 10px;

    @media (min-width: 1024px) {
        --spacing: 15px;
    }

    & > * {
        margin: var(--spacing);
        flex-shrink: 0;
    }

    &:before {
        content: '';
        margin-left: 20px;
    }

    &:after {
        content: '';
        margin-right: 20px;
    }
`;

const Storyboard = ({ data, showGallery, deleteCard, addToCaption }) => (
    <FlexRow>
        {data.map(({ src, alt, caption }, i) => (
            <Fragment key={i}>
                <Adder showGallery={showGallery} index={i} />
                <StoryboardImage
                    src={src}
                    alt={alt}
                    caption={caption}
                    deleteCard={deleteCard}
                    index={i}
                    addToCaption={addToCaption}
                />
            </Fragment>
        ))}
        <Adder showGallery={showGallery} index={data.length} />
    </FlexRow>
);

Storyboard.propTypes = {
    addToCaption: PropTypes.func.isRequired,
    showGallery: PropTypes.func.isRequired,
    deleteCard: PropTypes.func.isRequired,
    data: PropTypes.arrayOf(
        PropTypes.shape({
            src: PropTypes.string,
            alt: PropTypes.string,
            caption: PropTypes.string,
        })
    ),
};

export default Storyboard;

