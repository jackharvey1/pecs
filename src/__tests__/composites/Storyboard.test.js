import React from 'react';
import renderer from 'react-test-renderer';
import Storyboard from '../../components/composites/Storyboard';

describe('Storyboard component', () => {
    it('renders as intended', () => {
        const wrapper = renderer.create(<Storyboard
            showGallery={() => null}
            deleteCard={() => null}
            addToCaption={() => null}
            data={[
                {
                    src: 'http://image.com/image.png',
                    alt: 'image',
                    text: 'Caption',
                }, {
                    src: 'http://image.com/image.png',
                    alt: 'image',
                    text: 'Caption',
                }, {
                    src: 'http://image.com/image.png',
                    alt: 'image',
                    text: 'Caption',
                },
            ]}
        />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});
