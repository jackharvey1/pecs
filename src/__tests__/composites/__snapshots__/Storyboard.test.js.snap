// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyboard component renders as intended 1`] = `
.c5 {
  width: 125px;
  margin-top: 12px;
  border: none;
  color: #333333;
  font-size: 16px;
  font-family: Helvetica,Arial,sans-serif;
  text-align: center;
  resize: none;
}

.c5:focus {
  outline: none;
}

.c4 {
  box-shadow: 0 2px 8px 0 rgba(0,0,0,.25);
  margin: auto;
  width: 125px;
  height: 125px;
}

.c2 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 24px;
  top: 0;
  right: 0;
  z-index: 0;
}

.c3:after {
  content: '';
  position: absolute;
  width: 20px;
  height: 20px;
  border-radius: 10px;
  background-color: white;
  z-index: -1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 50px;
  width: 50px;
  font-size: 40px;
  border-style: dashed;
  border-color: lightgray;
}

.c1:first-child {
  margin-left: auto;
}

.c1:last-child {
  margin-right: auto;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  overflow-x: auto;
  overflow-y: hidden;
  width: 100vw;
  height: calc(100vh - 40px);
  --spacing: 10px;
}

.c0 > * {
  margin: var(--spacing);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0:before {
  content: '';
  margin-left: 20px;
}

.c0:after {
  content: '';
  margin-right: 20px;
}

@media (min-width:1024px) {
  .c5 {
    width: 150px;
  }
}

@media print {
  .c5::-webkit-input-placeholder {
    color: transparent;
  }

  .c5::-moz-placeholder {
    color: transparent;
  }

  .c5:-ms-input-placeholder {
    color: transparent;
  }

  .c5::placeholder {
    color: transparent;
  }
}

@media (min-width:1024px) {
  .c4 {
    width: 150px;
    height: 150px;
  }
}

@media print {
  .c3 {
    display: none;
  }
}

@media print {
  .c1 {
    display: none;
  }
}

@media (min-width:1024px) {
  .c0 {
    --spacing: 15px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <svg
      className={undefined}
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <svg
        className={undefined}
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
    <img
      alt="image"
      className="c4"
      src="http://image.com/image.png"
    />
    <textarea
      className="c5"
      cols="10"
      id="caption"
      onChange={[Function]}
      placeholder="Add caption here"
      rows="3"
      value={undefined}
    />
  </div>
  <div
    className="c1"
    onClick={[Function]}
  >
    <svg
      className={undefined}
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <svg
        className={undefined}
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
    <img
      alt="image"
      className="c4"
      src="http://image.com/image.png"
    />
    <textarea
      className="c5"
      cols="10"
      id="caption"
      onChange={[Function]}
      placeholder="Add caption here"
      rows="3"
      value={undefined}
    />
  </div>
  <div
    className="c1"
    onClick={[Function]}
  >
    <svg
      className={undefined}
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <svg
        className={undefined}
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
    <img
      alt="image"
      className="c4"
      src="http://image.com/image.png"
    />
    <textarea
      className="c5"
      cols="10"
      id="caption"
      onChange={[Function]}
      placeholder="Add caption here"
      rows="3"
      value={undefined}
    />
  </div>
  <div
    className="c1"
    onClick={[Function]}
  >
    <svg
      className={undefined}
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </div>
</div>
`;
